### PR TITLE
Fix garbage collector when using license manager from JuMP

### PR DESCRIPTION
**Pull request by frapac**
Current version of Knitro does not ensure that all Knitro's models are freed before releasing the license manager. It implies unsafe behavior when using JuMP (such as segfault). 

@kaarthiksundar:This PR adds official supports to use Knitro's license manager inside JuMP.

### CODE REVIEW

File: examples/jump_licensemanager.jl [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    @variable(model, 1 <= x[i=1:4] <= 5, start=initval[i])
+    @NLobjective(model, Min, x[1] * x[4] * (x[1] + x[2] + x[3]) + x[3])
+    c1 = @NLconstraint(model, x[1] * x[2] * x[3] * x[4] >= 25)
+    c2 = @NLconstraint(model, sum(x[i]^2 for i=1:4) == 40)
+
+    JuMP.optimize!(model)
+    @test JuMP.termination_status(model) == MOI.LOCALLY_SOLVED
+
+    # Warning! Free Knitro model before freeing license manager!
+    MOI.empty!(model.moi_backend)
```
Comment (mlubin) [inline examples/jump_licensemanager.jl]: We should not be providing examples where you have to access internal members of the JuMP model. We should have a proper API for this.

Comment (odow) [reply examples/jump_licensemanager.jl]: > We should have a proper API for this.

We do: `backend(model)`

File: examples/jump_licensemanager.jl [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+end
+
+# Free license manager
+KNITRO.KN_release_license(lm)
```
Comment (odow) [inline examples/jump_licensemanager.jl]: Can't the license manager store a list of models that are dependent on it, and either finalize them or throw a warning? (See CPLEX for an example)

File: examples/jump_licensemanager.jl (hunk 1 above)
Comment (frapac) [reply examples/jump_licensemanager.jl]: You are right. I will fix that by doing: 
```julia
MOI.empty!(model.moi_backend) 
```
(unless there exists a proper function in JuMP to free the backend)

File: examples/jump_licensemanager.jl (hunk 2 above)
Comment (frapac) [reply examples/jump_licensemanager.jl]: That is why the function `attach!` was designed for (see [here](https://github.com/JuliaOpt/KNITRO.jl/pull/103/files#diff-38e08dec1c54bf5d8d68fab6a7f9db8dR113)). The license manager keeps in memory to the Knitro's `Model` depending on it. When the license manager is released, we free them before releasing the license through the C API.

**Pull request merged**